src/Initializer/BatchRecorders/LocalIntegrationRecorder.cpp

```diff
@@ -145,24 +145,36 @@ void LocalIntegrationRecorder::recordLocalFluxIntegral() {
 
 
 void LocalIntegrationRecorder::recordDisplacements() {
-  real **displacements = currentLayer->var(currentHandler->displacements);
-  std::vector<real *> iVelocitiesPtrs{};
-  std::vector<real *> displacementsPtrs{};
+  real *(*faceDisplacements)[4] = currentLayer->var(currentHandler->faceDisplacements);
+  std::array<std::vector<real *>, 4> iVelocitiesPtrs{{}};
+  std::array<std::vector<real *>, 4> displacementsPtrs{};
+
+  constexpr unsigned numStressComponents{6};
+  constexpr unsigned offsetToVelocities = numStressComponents * (init::I::Stop[0] - init::I::Start[0]);
```

Comment:
There should be a better way to compute the tensor stride of the 1st dimension
```
init::I::Stop[0] - init::I::Start[0]
``` 
Please, suggest something

Reply:
Hm, I think this is fine. Don't know a better way right now